src/Daemon/BaseDaemon.cpp

```diff
@@ -134,6 +134,8 @@ static void terminateRequestedSignalHandler(int sig, siginfo_t *, void *)
 }
 
 
+static std::atomic<bool> fatal_error_printed{false};
```

Comment:
It's not quite correct to use std::atomic in a signal handler, it would be better to use `std::atomic_flag`. However, I doubt someone actually uses ClickHouse on architectures where it matters.